From 2688feb2742a6021ca211ae5c106b12c3d822946 Mon Sep 17 00:00:00 2001
From: Danny Milosavljevic <dannym@scratchpost.org>
Date: Mon, 11 Feb 2019 16:07:12 +0100
Subject: [PATCH] codec: vpx: Detect libvpx 1.8.0 and, if detected, use fewer
 frame formats in the chroma_table

Signed-off-by: Steve Lhomme <robux4@ycbcr.xyz>
(cherry picked from commit 5575fe3eb3fd46bada8662268b74d03493476a84)
---
 modules/codec/vpx.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/modules/codec/vpx.c b/modules/codec/vpx.c
index 2b1e37818d..aba180361f 100644
--- a/modules/codec/vpx.c
+++ b/modules/codec/vpx.c
@@ -116,6 +116,7 @@ static const struct
     { VLC_CODEC_I440, VPX_IMG_FMT_I440, 8, 0 },
 
     { VLC_CODEC_YV12, VPX_IMG_FMT_YV12, 8, 0 },
+#if VPX_IMAGE_ABI_VERSION < 5
     { VLC_CODEC_YUVA, VPX_IMG_FMT_444A, 8, 0 },
     { VLC_CODEC_YUYV, VPX_IMG_FMT_YUY2, 8, 0 },
     { VLC_CODEC_UYVY, VPX_IMG_FMT_UYVY, 8, 0 },
@@ -128,7 +129,7 @@ static const struct
 
     { VLC_CODEC_ARGB, VPX_IMG_FMT_ARGB, 8, 0 },
     { VLC_CODEC_BGRA, VPX_IMG_FMT_ARGB_LE, 8, 0 },
-
+#endif
     { VLC_CODEC_GBR_PLANAR, VPX_IMG_FMT_I444, 8, 1 },
     { VLC_CODEC_GBR_PLANAR_10L, VPX_IMG_FMT_I44416, 10, 1 },
 
commit 41caaa08cde60c4fec4bf2e5f9610e2a1b9e6a23
Author: Vinson Lee <vlee@freedesktop.org>
Date:   Thu Feb 5 14:48:53 2015 -0800

    lua: Define LUA_COMPAT_APIINTCASTS for Lua >= 5.3.0 compatibility.
    
    In Lua 5.3.0, luaL_checkint was deprecated.
    
    This patch fixes this build error with Lua 5.3.0.
    
    lua/demux.c: In function ‘vlclua_demux_peek’:
    lua/demux.c:55:5: error: implicit declaration of function ‘luaL_checkint’ [-Werror=implicit-function-declaration]
         int n = luaL_checkint( L, 1 );
         ^
    
    Signed-off-by: Vinson Lee <vlee@freedesktop.org>
    Signed-off-by: Jean-Baptiste Kempf <jb@videolan.org>

diff --git a/modules/lua/vlc.h b/modules/lua/vlc.h
index efd94f1..85c7fc1 100644
--- a/modules/lua/vlc.h
+++ b/modules/lua/vlc.h
@@ -38,6 +38,7 @@
 #include <vlc_stream.h>
 #include <vlc_demux.h>
 
+#define LUA_COMPAT_APIINTCASTS
 #define LUA_COMPAT_MODULE
 #include <lua.h>        /* Low level lua C API */
 #include <lauxlib.h>    /* Higher level C API */
